File: resource-registry/src/test/java/nl/haploid/octowight/registry/data/ResourceRootTest.java
package nl.haploid.octowight.registry.data;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ResourceRootTest {

	@Test
	public void testKey() {
		final ResourceRoot resourceRoot = new ResourceRoot();
		resourceRoot.setResourceType("cosgrove");
		resourceRoot.setAtomId(69l);
		resourceRoot.setAtomOrigin("california");
		resourceRoot.setAtomType("campbell");
		final String expectedKey = "california:campbell/69->cosgrove";
		final String actualKey = resourceRoot.key();
		assertEquals(expectedKey, actualKey);
	}
}

File: resource-registry/src/test/scala/nl/haploid/octowight/registry/data/ResourceRootTest.scala
package nl.haploid.octowight.registry.data

import nl.haploid.octowight.registry.AbstractTest

class ResourceRootTest extends AbstractTest {

  behavior of "Resource root"

  it should "return a key" in {
    val atom = new Atom(69l, "california", "campbell")
    val resourceRoot = new ResourceRoot(resourceId = null, resourceCollection = "cosgrove", root = atom, version = null)
    val expectedKey = "california:campbell/69->cosgrove"
    val actualKey = resourceRoot.key
    actualKey should be(expectedKey)
  }
}
